### PR TITLE
Add AZ name to subnet name

### PR DESCRIPTION
**Pull request by s2504s**
## What
* Add AZ name to subnet name

## Why
* We want to have comprehensive subnet names

### CODE REVIEW

File: private.tf [hunk 1]
```diff
@@ -8,12 +8,23 @@ module "private_label" {
...
+}
+
 resource "aws_subnet" "private" {
   count             = "${length(var.availability_zones)}"
   vpc_id            = "${data.aws_vpc.default.id}"
   availability_zone = "${element(var.availability_zones, count.index)}"
   cidr_block        = "${cidrsubnet(signum(length(var.cidr_block)) == 1 ? var.cidr_block : data.aws_vpc.default.cidr_block, ceil(log(length(data.aws_availability_zones.available.names) * 2, 2)), count.index)}"
-  tags              = "${module.private_label.tags}"
+  tags              = {
+    "Name"          = "${module.private_subnet_label.id}-${element(var.availability_zones, count.index)}"
```
Comment (comeanother) [inline private.tf]: Let's replace `-` with `var.delimiter` and `-` sign occurrences in `element(var.availability_zones, count.index)` to `var.delimiter`.

File: public.tf [hunk 2]
```diff
@@ -13,7 +20,11 @@ resource "aws_subnet" "public" {
   vpc_id            = "${data.aws_vpc.default.id}"
   availability_zone = "${element(var.availability_zones, count.index)}"
   cidr_block        = "${cidrsubnet(signum(length(var.cidr_block)) == 1 ? var.cidr_block : data.aws_vpc.default.cidr_block, ceil(log(length(data.aws_availability_zones.available.names) * 2, 2)), length(data.aws_availability_zones.available.names) + count.index)}"
-  tags              = "${module.public_label.tags}"
+  tags              = {
+    "Name"          = "${module.public_subnet_label.id}-${element(var.availability_zones, count.index)}"
```
Comment (comeanother) [inline public.tf]: Let's replace `-` with `var.delimiter` and `-` sign occurrences in `element(var.availability_zones, count.index)` to `var.delimiter`.

Comment (osterman) [reply public.tf]: wow this was a lot trickier than expected

**Comment by comeanother** [comment]
Tested.
screenshot from 2017-09-19 16-32-42

AWS console:
screenshot from 2017-09-19 16-33-01

**Pull request merged**